bot/cogs/bot.py

```diff
@@ -78,32 +80,35 @@ def __init__(self, bot: AutoShardedBot):
             if time.time()-self.previous_format_times[msg.channel.id] > 300 or msg.channel.id == DEVTEST_CHANNEL:
                 if msg.content.count("\n") >= 3:
                     try:
-                        # Some users formatted multi-line code using `'sip("
-                        # This fixes that by treating them as nothing.
-                        if "`" in msg.content:
-                            msg.content = msg.content[:-1][1:]
-                        tree = ast.parse(msg.content)
+                        # Filtering valid Python codeblocks and exiting if a valid Python codeblock is found
+                        if re.match(msg.content, "```(python|py)\n((?:.*\n*)+)```", re.IGNORECASE):
+                            return
+                        else:
+                            # Format the block into python source
+                            content = ""
+                            for line in msg.content.split("\n"):
+                                content += line.strip("`") + "\n"
+                            # Remove "Python" or "Py" from top of the message if exists
+                            if content.startswith("python") or content.startswith("Python"):
+                                content = content[6:]
+                            elif content.startswith("py") or content.startswith("Py"):
+                                content = content[2:]
+                            # Strip again to remove the whitespace(s) left if the msg looks like
+                            # " Python"
```

Comment:
this comment doesn't make sense considering you've just removed "python" from content if it exists. The whitespace you're removing is the one _after_ python but before the code if the user had typed this:

\`\`\`
python if condition:
\`\`\`